### DISCUSSION TITLE
How to handle commas in Budibase numeric field entry

### DISCUSSION THREAD

**Discussion by mmays77** [Budibase/budibase#12787] [Help] [answered]
I am using Budibase community version 2.13.50 on Ubuntu 22.04. The backend is Mysql 8.0.32. I'm new to Budibase but have solved a lot of problems with it and like the environment.

I created a data entry screen that requires entry of currency values. These values are represented as numeric in Mysql, and in the Budibase entry form the form field is numeric. The input data is most often copied and pasted from other websites, and the numbers sometimes have commas in them. 

When a numeric value containing a comma is pasted into a numeric field, when "save" is pressed, the system appears to not have accepted the numeric value, and an error is produced saying the value is required. If the user manually deletes any commas in the input field, then the data is accepted and the save works properly. 

Using a handlebar "remove" expression, or a "replace" expression, or a javascript replace(/,/g,'') triggered by an "on change" event for the field, do not work because the system does not recognize that there is any data in the numeric field as long as there is a comma in it.

If I do the same thing with an alphanumeric text entry field, the "on change" event behavior of the remove and replace expressions work as expected, but then the system gives an error that says that a text field cannot be used to save data to a numeric database column.

I have tried putting "update field value" actions on the "save" button but they don't work because the numeric field thinks there is no data in it as long as there is a comma in the entered data.

I'm at a loss here trying to find a work-around. It is inconvenient for the users to manually delete the commas every time they paste the data, and I can't think of a way to have Budibase functionality ignore, delete or strip the commas out of the numeric data entry field.

Any suggestions would be greatly appreciated. Thanks in advance!!

**Comment by ConorWebb96** [comment]
Hey @mmays77,

The simplest solution would be to just use a text column and then the text field type within the form.

A more complicated fix would be to use a dummy text field which can be copied into regardless of formatting. From here with the on-change action you could then use some regex JS to convert this to an acceptable number for the now hidden actual number field.

The function below is what I used to achieve this inside the dummy number text field .

```js
var inputString = $("Field Value");
var formattedString = inputString.replace(/[^0-9]/g, '');
var formattedNumber = parseInt(formattedString, 10);

return formattedNumber;
```
<img width="1512" alt="Screenshot 2024-01-16 at 10 11 39" src="https://github.com/Budibase/budibase/assets/126772285/716fb183-2b8e-41a9-b41b-ee736455d143">
<img width="1512" alt="Screenshot 2024-01-16 at 10 12 00" src="https://github.com/Budibase/budibase/assets/126772285/16404216-3eb3-40a0-95c3-7fe66f7ecd77">

Heres a quick gif of this in action.
![20240116_101313](https://github.com/Budibase/budibase/assets/126772285/1af56f0b-738d-4171-99b0-5653381621f0)

**Reply by melohagan** [reply]
There's also an input mask plugin component that may be useful for a currency field: https://github.com/melohagan/budibase-component-nikita-input-mask

**Reply by melohagan** [reply, marked as answer]
If you need to save a numeric value and not a string, then you can do that in your [Save Row](https://docs.budibase.com/docs/data-actions#save-row) action. Click `Add column`, select the numeric column of your SQL table, then use the binding to cast the text field to a number.

**Comment by mmays77** [comment]
Thanks to both of you for your help and your responses. The part that I missed was the binding of fields to database columns in the "save row" action. I'm back on track now. 

Thanks again!
